static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js

```diff
@@ -0,0 +1,100 @@
+    'bonzo',
+    'fastdom',
+    'common/utils/_',
+    'common/utils/$',
+    'common/utils/detect',
+    'common/utils/mediator',
+    'common/utils/storage',
+    'common/utils/template',
+    'common/views/svgs',
+    'text!common/views/commercial/creatives/expandable-video-v2.html'
+], function (
+    bean,
+    bonzo,
+    fastdom,
+    _,
+    $,
+	detect,
+    mediator,
+    storage,
+    template,
+    svgs,
+    ExpandableVideoTpl
+) {
+
+    /**
+     * https://www.google.com/dfp/59666047#delivery/CreateCreativeTemplate/creativeTemplateId=10028247
+     */
+    var ExpandableVideo = function ($adSlot, params) {
+        this.$adSlot      = $adSlot;
+        this.params       = params;
+        this.isClosed     = true;
+
+        if (detect.isBreakpoint({min: 'tablet'})) {
+            this.closedHeight = 250;
+            this.openedHeight = 500;
+        } else {
+            this.closedHeight = 150;
+            this.openedHeight = 300;
+        }
+
+    };
+
+    ExpandableVideo.prototype.create = function () {
+        var videoHeight = this.openedHeight,
+            showmoreArrow = {
+                showArrow: (this.params.showMoreType === 'arrow-only' || this.params.showMoreType === 'plus-and-arrow') ?
+                    '<button class="ad-exp__open-chevron ad-exp__open">' + svgs('arrowdownicon') + '</button>' : ''
+            },
+            showmorePlus = {
+                showPlus: (this.params.showMoreType === 'plus-only' || this.params.showMoreType === 'plus-and-arrow') ?
+                    '<button class="ad-exp__close-button ad-exp__open">' + svgs('closeCentralIcon') + '</button>' : ''
+            },
+            videoSource = {
+                videoEmbed: (this.params.YoutubeVideoURL !== '') ?
+                    '<iframe id="YTPlayer" width="100%" height="' + videoHeight + '" src="' + this.params.YoutubeVideoURL + '?showinfo=0&amp;rel=0&amp;controls=0&amp;fs=0&amp;title=0&amp;byline=0&amp;portrait=0" frameborder="0" class="expandable-video"></iframe>' : ''
+            },
+            $ExpandableVideo = $.create(template(ExpandableVideoTpl, { data: _.merge(this.params, showmoreArrow, showmorePlus, videoSource) })),
+            domPromise = new Promise(function (resolve) {
+                fastdom.write(function () {
+
+                    this.$ad = $('.ad-exp--expand', $ExpandableVideo).css('height', this.closedHeight);
+
+                    $('.ad-exp-collapse__slide', $ExpandableVideo).css('height', this.closedHeight);
+
+                    if (this.params.trackingPixel) {
+                        this.$adSlot.before('<img src="' + this.params.trackingPixel + this.params.cacheBuster + '" class="creative__tracking-pixel" height="1px" width="1px"/>');
+                    }
+                    $ExpandableVideo.appendTo(this.$adSlot);
+                    resolve();
+                }.bind(this));
+            }.bind(this));
+
+        bean.on(this.$adSlot[0], 'click', '.ad-exp__open', function () {
```

Comment:
I'm not sure if the relationship between the event and the script is totally transparent. I'm a big fan of using data attributes and bindings - or at least 'js-' prefixed classes - to mark elements that will handle and invoke events. This makes it harder to break the scripted behaviour when trying to modify the HTML. Just my opinion.